test/jdk/java/util/HexFormat/HexFormatTest.java

```diff
@@ -0,0 +1,463 @@
+                }
+            }
+        }
+    }
+
+    @Test(dataProvider="HexFormattersParsers")
+    static void testDecoderString(String unused1, String unused2, String unused3,
+                                   boolean unused4, HexFormat hex) {
+        byte[] expected = genBytes('A', 15);
+        String s = hex.formatHex(expected);
+        System.out.println("    formatted: " + s);
+
+        byte[] actual = hex.parseHex(s);
+        System.out.println("    parsed as: " + Arrays.toString(expected));
+        int mismatch = Arrays.mismatch(expected, actual);
+        Assert.assertEquals(actual, expected, "encode/decode cycle failed, mismatch: " + mismatch);
+    }
+
+    @Test(dataProvider="HexFormattersParsers")
+    static void testDecoderCharArray(String unused1, String unused2, String unused3,
+                                     boolean unused4, HexFormat hex) {
+        byte[] expected = genBytes('A', 15);
+        String s = hex.formatHex(expected);
+        System.out.println("    formatted: " + s);
+
+        char[] chars = s.toCharArray();
+        byte[] actual = hex.parseHex(chars, 0, chars.length);
+        System.out.println("    parsed as: " + Arrays.toString(expected));
+        int mismatch = Arrays.mismatch(expected, actual);
+        Assert.assertEquals(actual, expected, "format/parse cycle failed, mismatch: " + mismatch);
+    }
+
+    @Test(dataProvider="HexFormattersParsers")
+    static void testFormatterToString(String delimiter, String prefix, String suffix,
+                                    boolean uppercase,
+                                    HexFormat hex) {
+        String actual = String.format(
+                "uppercase: %s, delimiter: \"%s\", prefix: \"%s\", suffix: \"%s\"",
+                uppercase, escapeNL(delimiter), escapeNL(prefix), escapeNL(suffix));
+        System.out.println("    hex: " + actual);
+        Assert.assertEquals(actual, hex.toString(), "Formatter toString mismatch");
+    }
+
+    private static String escapeNL(String string) {
+        return string.replace("\n", "\\n")
+                .replace("\r", "\\r");
+    }
+
+    @Test
+    static void testVariableLength() {
+        HexFormat hex = HexFormat.of();
+
+        final long orig = 0xfedcba9876543210L;
+        for (int digits = 0; digits <= 16; digits++) {
+            String s = hex.toHexDigits(orig, digits);
+            long actual = hex.fromHexDigitsToLong(s, 0, digits);
+            System.out.printf("    digits: %2d, formatted: \"%s\", parsed as: 0x%016xL%n",
+                    digits, s, actual);
+        }
+    }
+
+    @Test
+    private static void samples() {
+
+        {
+            // Primitive formatting and parsing.
+            HexFormat hex = HexFormat.of();
+
+            byte b = 127;
+            String byteStr = hex.toHexDigits(b);
+            System.out.println("    " + byteStr);
+
+            int byteVal = hex.fromHexDigits(byteStr);
+            assert(byteStr.equals("7f"));
```

Comment:
These code fragments are the examples used in the javadoc; making a simple copy/paste from working code to the javadoc.
In the example code `assert`, a Java language construct is cleaner than showing testng assertions).

Reply:
Yes using test framework specific code in the example code is probably not a good idea and the `assert` statement there makes the intention clear.

However, for unit tests I assume it can be a little bit problematic if the tests are compiled with assertions disabled. Additionally if one of the `assert` statements fail you would only get an `AssertionError` without much additional feedback, while test framework assertion methods usually provide pretty detailed messages and have good IDE support. For example `assertEquals("a", "b")` would yield something like
> Expected "a" but was "b"

which makes it obvious what the issue is.